ui/src/hooks/functions.tsx

```diff
@@ -4,106 +4,147 @@ import {
   useQueryClient,
   UseQueryOptions,
 } from '@tanstack/react-query';
-import apiClient from '../context/client';
+import { CLIENT } from '../context/catalog';
+import { route } from '../utils/openapi';
+import type {
+  paths as CatalogApi,
+  components as CatalogComponent,
+} from '../types/api/catalog.gen';
+import type {
+  ApiInterface,
+  ApiSuccessResponse,
+  ApiRequestPathParam,
+  ApiRequestQueryParam,
+} from '../utils/openapi';
 
-export interface FunctionInterface {
-  function_id: string;
-  catalog_name: string;
-  schema_name: string;
-  name: string;
-  comment: string;
-  external_language: string;
-  routine_definition: string;
-  created_at: number;
-  updated_at: number | null;
-  owner: string | null;
-  created_by: string | null;
-  updated_by: string | null;
-}
-interface ListFunctionsResponse {
-  functions: FunctionInterface[];
-  next_page_token: string | null;
-}
+export type FunctionInterface = ApiInterface<CatalogComponent, 'FunctionInfo'>;
 
-interface ListFunctionsParams {
-  catalog: string;
-  schema: string;
+// TODO:
+// The queries `max_results` and `page_token` are not properly handled as of [25/11/2024].
+// These queries need to be implemented.
+export type UseListFunctionsArgs = ApiRequestQueryParam<
+  CatalogApi,
+  '/functions',
+  'get'
+> & {
   options?: Omit<
-    UseQueryOptions<ListFunctionsResponse>,
+    UseQueryOptions<ApiSuccessResponse<CatalogApi, '/functions', 'get'>>,
     'queryKey' | 'queryFn'
   >;
-}
+};
 
 export function useListFunctions({
-  catalog,
-  schema,
+  catalog_name,
+  schema_name,
+  max_results,
+  page_token,
   options,
-}: ListFunctionsParams) {
-  return useQuery<ListFunctionsResponse>({
-    queryKey: ['listFunctions', catalog, schema],
+}: UseListFunctionsArgs) {
+  return useQuery<ApiSuccessResponse<CatalogApi, '/functions', 'get'>>({
+    queryKey: ['listFunctions', catalog_name, schema_name],
     queryFn: async () => {
-      const searchParams = new URLSearchParams({
-        catalog_name: catalog,
-        schema_name: schema,
+      const api = route(CLIENT, {
+        path: '/functions',
+        method: 'get',
+        params: {
+          query: {
+            catalog_name,
+            schema_name,
+            max_results,
+            page_token,
```

Comment:
Not sure these are needed since we are only doing client side pagination at this time